bin/varnishd/cache/cache_vrt.c

```diff
@@ -93,14 +93,6 @@ VRT_synth(VRT_CTX, VCL_INT code, VCL_STRING reason)
 		return;
 	}
 
-	if (reason && !WS_Allocated(ctx->ws, reason, -1)) {
-		reason = WS_Copy(ctx->ws, reason, -1);
-		if (!reason) {
-			VRT_fail(ctx, "Workspace overflow");
-			return;
-		}
-	}
```

Comment:
This might not be safe actually:

```vcl
return (synth(200, some_vmod.some_string()));
```

In that case we can't assume a lifetime beyond the `VRT_synth()` call for the `reason` argument.

Reply:
see https://github.com/varnishcache/varnish-cache/pull/4269#issuecomment-2663362814 regarding this and the next four review comments.